### PR TITLE
oleh_tests

### PR DESCRIPTION
**Pull request by OlegRuin84**
This merge was done after added eslint and wrote tests.

### CODE REVIEW

File: src/components/button/Button.module.scss [hunk 1]
```diff
@@ -11,4 +11,4 @@
     color: #fff;
     font-size: variables.$fs-s;
     font-weight: 700;
-}
\ No newline at end of file
+}
```
Comment (OlegRuin84) [inline src/components/button/Button.module.scss:14]: interval

File: src/components/button/Button.jsx [hunk 2]
```diff
@@ -13,7 +21,7 @@ export default function Button({ text="", color="rgba(70, 163, 88, 1)", padding=
 
 
   return  (
-    <Link to={toPage} style={buttonStyle} className={styles.buttonStyle} onClick={onClick} {...rest}>
+    <Link to={toPage} style={buttonStyle} className={styles.buttonStyle} onClick={onClick} {...rest}  data-testid='button'>
```
Comment (OlegRuin84) [inline src/components/button/Button.jsx:24]: add data-testid

File: package.json [hunk 3]
```diff
@@ -53,6 +53,7 @@
   },
   "devDependencies": {
     "sass": "^1.68.0",
-    "scss": "^0.2.4"
+    "scss": "^0.2.4",
+    "@babel/plugin-proposal-private-property-in-object": "^7.18.9"
```
Comment (OlegRuin84) [inline package.json:57]: add one string

File: src/components/card/Icons.jsx [hunk 4]
```diff
@@ -68,7 +68,7 @@ export function Icons({ itemNo, name, price, imageURL, id }) {
 
 
   return (
-    <div className={styles.cardItemIconsWrapper}>
+    <div className={styles.cardItemIconsWrapper} data-testid = 'icons-testid'>
```
Comment (OlegRuin84) [inline src/components/card/Icons.jsx:71]: add data-testid

File: src/components/cardlists/AllCategoriesCardList.jsx [hunk 5]
```diff
@@ -62,7 +62,7 @@ export default function CategoriesCardList() {
   
   return (
     <>
-      <div className={styles.filtrationWrapper}>
+      <div className={styles.filtrationWrapper} data-testid='allCategoriesCardList-test'>
```
Comment (OlegRuin84) [inline src/components/cardlists/AllCategoriesCardList.jsx:65]: add data-testid

File: src/components/cardlists/AllCategoriesCardList.test.jsx
Comment (OlegRuin84) [inline src/components/cardlists/AllCategoriesCardList.test.jsx:1]: add new test

File: src/components/cardlists/FilteredCardList.jsx [hunk 6]
```diff
@@ -45,7 +45,7 @@ export default function FilteredCardList( {property, value, priceRange} ) {
     
 
     return (
-        <CardList items={items}/>
+      <CardList items={items} data-testid='filteredCardList-test'/>
```
Comment (OlegRuin84) [inline src/components/cardlists/FilteredCardList.jsx:48]: add data-testid

File: src/components/categoriesList/CategoriesList.jsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ import Auction from "./icons/lots/Auction"
 const Categorys = () => {
 
 return (
-  <section className={styles.categorysWrapper}>
+  <section className={styles.categorysWrapper} data-testid='categoriesList'>
```
Comment (OlegRuin84) [inline src/components/categoriesList/CategoriesList.jsx:12]: ad data-testid

File: src/components/categoriesList/CategoriesList.test.jsx
Comment (OlegRuin84) [inline src/components/categoriesList/CategoriesList.test.jsx:1]: create tests for CategiriesList

File: src/components/footer/Footer.test.jsx [hunk 8]
```diff
@@ -1,12 +1,23 @@
 import Footer from './Footer';
 import {render,screen} from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
```
Comment (OlegRuin84) [inline src/components/footer/Footer.test.jsx:3]: add BrowserRouter

File: src/components/footer/Footer.test.jsx [hunk 9]
```diff
@@ -1,12 +1,23 @@
...
 
-describe('should the Footer', () =>{
 
-    test('should render the Footer', () =>{
-        render (<Footer />)
+jest.mock("axios", () => ({
+  post: jest.fn(),
+}));
```
Comment (OlegRuin84) [inline src/components/footer/Footer.test.jsx:8]: mocking axios

File: src/components/footer/Footer.test.jsx [hunk 10]
```diff
@@ -1,12 +1,23 @@
...
+
+
+describe('should the Footer', () =>{
```
Comment (OlegRuin84) [inline src/components/footer/Footer.test.jsx:11]: create testes

File: src/components/footer/Footer.test.jsx [hunk 11]
```diff
@@ -1,12 +1,23 @@
...
 
-        const footer = screen.getByTestId('footer-svg-test');
-        expect(footer.getAttribute('data-testid')).toBe('footer-svg-test');
-    });
-});
\ No newline at end of file
+  test('should render the Footer', () =>{
+    render (
```
Comment (OlegRuin84) [inline src/components/footer/Footer.test.jsx:14]: render component whith BrowserRouter

File: src/components/footer/Footer.test.jsx [hunk 12]
```diff
@@ -1,12 +1,23 @@
...
+      <BrowserRouter>
+        <Footer />
+      </BrowserRouter>
+    )
+    
+    const footer = screen.getByTestId('footer-svg-test');
```
Comment (OlegRuin84) [inline src/components/footer/Footer.test.jsx:20]: search data-testId

File: src/components/footer/icons/linkedin/Linkedin.test.jsx [hunk 13]
```diff
@@ -1,4 +1,4 @@
-import Linkedin from './Linkedin';
+import Linkedin from './LinkedIn';
```
Comment (OlegRuin84) [inline src/components/footer/icons/linkedin/Linkedin.test.jsx:1]: import component

File: src/components/routes/adminRoutes/adminPage/AdminPage.test.jsx
Comment (OlegRuin84) [inline src/components/routes/adminRoutes/adminPage/AdminPage.test.jsx:1]: chench test

**Review by OlegRuin84 — COMMENTED** [review]
i look this change and i say - "Yes"

**Review by grishanych — APPROVED** [review]

**Review by grishanych — APPROVED** [review]
all conflicts were resolved

**Pull request closed without merging**